eNMS/controller/inventory.py

```diff
@@ -94,16 +95,25 @@ def handoffssh(self, id, **kwargs):
             device=device.id,
         )
         Session.commit()
-        Thread(
-            target=SshConnection,
-            args=(device.ip_address, *credentials, session.id, uuid, port),
-        ).start()
-        return {
-            "port": port,
-            "username": uuid,
-            "device_name": device.name,
-            "device_ip": device.ip_address,
-        }
+        try:
+            ssh_conn = SshConnection(
+                device.ip_address, *credentials, session.id, uuid, port
+            )
+        except Exception as e:
+            ssh_conn = e
+        if isinstance(ssh_conn, SshConnection):
+            Thread(
+                target=SshConnection.start_session,
+                args=(ssh_conn, session.id, uuid, port),
+            ).start()
+            return {
+                "port": port,
+                "username": uuid,
+                "device_name": device.name,
```

Comment:
My understanding is that if the `try` block works, `ssh_conn` will be an instance of Ssh connection so you'll enter the `if` block later, and if it doesn't, it will be an exception and you'll end up in the `else` block. So why not just put the `if` block logic directly in the `try` block, and in the `except` block you just return the dictionary of the `else` block. It's the same in the end but it would feel less convoluted

Reply:
I think I see what you're getting at.  I'll play with that and see if I knock it out.

Reply:
See if this handles things.  I've tested a successful, bad credentials, and time out scenario and things are behaving as expected.